src/lib/comms/sol-coap.c

```diff
@@ -1295,6 +1299,10 @@ respond_packet(struct sol_coap_server *server, struct sol_coap_packet *req,
         return cb(server, resource, req, cliaddr, (void *)c->data);
     }
 
+    if (server->unknown_handler)
+        return server->unknown_handler(server, req, cliaddr,
+            server->unknown_handler_data);
```

Comment:
I think I will change this one for now and later I come back and change the whole coap.